src/requirements/requirement-frontend-utils.ts

```diff
@@ -247,15 +251,40 @@ type MatchedRequirementFulfillmentSpecification =
 
 /**
  * The function respects the user choice on toggleable requirement, and provides the already decided
- * fulfillment strategy to follow.
+ * fulfillment strategy to follow. It also filters eligible courses based on the user's college/major.
  *
  * @returns a spec telling how the requirement progress should be computed, or null if the requirement
  * is self-check.
  */
 export function getMatchedRequirementFulfillmentSpecification(
   requirement: RequirementWithIDSourceType,
+  { college }: AppOnboardingData,
   toggleableRequirementChoices: AppToggleableRequirementChoices
 ): MatchedRequirementFulfillmentSpecification {
+  const { sourceType, sourceSpecificName } = requirement;
+  const filterEligibleCoursesByRequirementConditions = (
+    coursesList: readonly (readonly number[])[],
+    conditions: RequirementCourseConditions | undefined
+  ) =>
+    coursesList.map(courses =>
+      courses.filter(courseId => {
+        // no requirement conditions
+        if (!(conditions && courseId in conditions)) return true;
+        const { colleges, majorsExcluded } = conditions[courseId];
+        // user college is in colleges list
+        if (college && !colleges.includes(college)) return false;
+        // requirement is in colleges list
+        if (sourceType === 'College' && colleges.includes(sourceSpecificName)) return true;
```

Comment:
Confused about these two different if statements. The second one makes sense to me as it is structured the same as the `majorsExcluded` check - but what purpose does the first one serve in checking `!colleges.includes(college)`? Is the second check not sufficient?

Reply:
Hmm, looking at it now I think it may need some reworking. I might need to tweak this a little bit.

Reply:
Does it make more sense now after my tweaks?

You're right, these two if statements are redundant. I think I only included it as an early return, but it should be the same logic without it. Right now, users can only be in one college. So, there's never a case where their college and college requirements are mismatched.